src/App.tsx

```diff
@@ -0,0 +1,187 @@
+} from "@mui/material";
+import SendIcon from "@mui/icons-material/Send";
+import React from "react";
+import CodeMirror from "@uiw/react-codemirror";
+import { oneDark } from "@codemirror/theme-one-dark";
+import init, { program } from "miden-wasm";
+import "./App.css";
+
+async function getExample(example: string[]) {
+    const inputs = fetch(`https://raw.githubusercontent.com/0xPolygonMiden/examples/main/examples/${example}.inputs`)
+    const masm = fetch(`https://raw.githubusercontent.com/0xPolygonMiden/examples/main/examples/${example}.masm`)
+    return [(await inputs).text(), (await masm).text()];
+  }
+
+function App() {
+
+  const [inputs, setInputs] = React.useState(
+    `{
+      "stack_init": ["0"],
+      "advice_tape": ["0"]
+}`
+  );
+
+  var exampleCode =`begin
+  push.1
+  push.2
+  add
+end`
+
+  const [code, setCode] = React.useState(exampleCode);
+
+  const [output, setOutput] = React.useState({
+    init: false,
+    success: true,
+    text: "",
+  });
+
+  const [numOfOutputs, setNumOfOutputs] = React.useState(1);
+
+  // this should be dyncamically read from ./examples/ - buy my TS knowledge is limited
+  const examples = [
+    'collatz',
+    'comparison',
+    'conditional',
+    'fibonacci',
+    'game-of-life-4x4',
+    'nPrime',
+  ];
+  
+ 
+  const [example, setExample] = React.useState<string[]>([]);
+  const handleChangeMultiple = async (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { options } = event.target;
+    const value: string[] = [];
+    for (let i = 0, l = options.length; i < l; i += 1) {
+      if (options[i].selected) {
+        value.push(options[i].value);
+      }
+    }
+    setExample(value)
+    
+    const example_code = await getExample(value)
+    
+    setInputs(await example_code[0])
+    setCode(await example_code[1])    
+  }
+
+  return (
+    <>
+      <AppBar position="static">
+        <Toolbar variant="dense">
+          <Typography variant="h6" color="inherit" component="div">
+            Playground for Miden Examples in Miden Assembly
+          </Typography>
+        </Toolbar>
+      </AppBar>
+      <Container maxWidth="xl">
+        <Box sx={{ my: 4 }}>
+          <FormControl variant="outlined" sx={{ width: '100%' }}>
+              <InputLabel shrink htmlFor="select-multiple-native">
+                Select Example
+              </InputLabel>
+              <Select
+                multiple
+                native
+                value={example}
+                // @ts-ignore Typings are not considering `native`
+                onChange={handleChangeMultiple}
+                label="Native"
+                inputProps={{
+                  id: 'select-multiple-native',
+                }}
+              >
+                {examples.map((name) => (
+                  <option key={name} value={name}>
+                    {name}
+                  </option>
+                ))}
+              </Select>
+          </FormControl>
+        </Box>
```

Comment:
We can improve this `Select` by following the MaterialUI `Select` component pattern. I also think we should move the outputs and execute button to the top and make a few minor adjustments to the styling. I propose:

1. Delete everything below the source code `CodeMirror`
2. Replace this highlighted box with the box code below.
3. Replace `handleChangeMultiple` with `handleSelectChange` code below.
4. Change the type of `example` and the input to `getExample` to `string` instead of `string[]`

```
<Box sx={{ display: 'flex', flexDirection: 'row', mx: 'auto', my: 4}}>
          <FormControl variant="outlined" sx={{minWidth: 120}}>
              <InputLabel shrink htmlFor="select-example">
                Example
              </InputLabel>
              <Select
                value={example}
                onChange={handleSelectChange}
                labelId='select-example'
              >
                {examples.map((name) => (
                  <MenuItem value={name} key={name}>{name}</MenuItem>
                ))}
              </Select>
          </FormControl>
          <FormControl variant="outlined" sx={{minWidth: 120, mx: 2}}>
              <InputLabel shrink htmlFor="select-outputs">
                Outputs
              </InputLabel>
              <Select
              value={numOfOutputs}
              onChange={(e) => setNumOfOutputs(Number(e.target.value))}
              labelId='select-outputs'
            >
              <MenuItem value={0} key={0}>0</MenuItem>
              <MenuItem value={1} key={1}>1</MenuItem>
              <MenuItem value={16} key={16}>16</MenuItem>
            </Select>
          </FormControl>
          <Button
            variant="contained"
            color="primary"
            onClick={() => {
              init().then(() => {
                try {
                  let resp = program(code, inputs, numOfOutputs);
                  console.log(resp);
                  setOutput({
                    init: true,
                    success: true,
                    text: resp.join(" "),
                  });
                } catch (error) {
                  setOutput({
                    init: true,
                    success: false,
                    text: "Error: Check the developer console for details.",
                  });
                }
              });
            }}
            endIcon={<SendIcon />}
          >
            Execute
          </Button>
          {output.init ? (
            <Alert
              variant="outlined"
              severity={output.success ? "success" : "error"}
              sx={{mx: 2}}
            >
              <Typography color="inherit" component="div">
                {output.text}
              </Typography>
            </Alert>
          ) : (
            <></>
          )}
        </Box>
```

```
  const handleSelectChange = async (event: SelectChangeEvent) => {
    const value = event.target.value;
    // set the current example to the selected one
    setExample(value)

    // retrieve the example data and update the inputs and code
    const example_code = await getExample(value)
    setInputs(await example_code[0])
    setCode(await example_code[1])    

    // reset the output
    setOutput(emptyOutput)
  }
```

Reply:
tbh, it actually probably makes sense to define this as its own component and then put it at both the top and the bottom for better UX on the long examples